test/blackbox-tests/test-cases/formatting/feature.t/run.t

```diff
@@ -3,6 +3,10 @@ Note about versioning:
 - in (lang dune 1.x), no formatting rules are set up by default.
 - (lang dune 2.0) behaves as if (using fmt 1.2) is set.
 
+Turn every symlinks into writable files belonging to the sandbox as formatting
+rules are not generated for read-only files:
+  $ find . -type l | while read f; do (rm "$f"; cat > $f) < $f; done
```

Comment:
Wouldn't this be easier with `find -exec`?

(OTOH fairly impressed it even works, reading the file and replacing it in one step)

Reply:
All the files (but not the directories) in the cram test's current directory are symlinks to read-only copies of the source code. This breaks `dune promote` but also breaks any programs that don't do the "write to temp file and move it to its destination" strategy, as opening the file for truncating and writing (eg. with `open_out_bin`) will not override the permissions.

`-exec` will not work because this uses two commands. Reading continue to work after the file is removed, after all the stdin is still a reference to it so it can't be freed yet. `cat > $f` outputs to a new file and doesn't disturb the reading.